cfme/tests/networks/test_sdn_addresses.py

```diff
@@ -0,0 +1,42 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.networks.views import FloatingIpView
+
+pytestmark = [
+    test_requirements.sdn,
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.provider([OpenStackProvider], scope="module")
+]
+
+
+@pytest.fixture()
+def network_manager(appliance, provider):
+    network_managers = appliance.collections.network_providers.filter({"provider": provider})
```

Comment:
```suggestion
    network_manager, = appliance.collections.network_providers.filter({"provider": provider}).all()
    yield network_manager
```

This way, when there are more than one network_provider, we will get an error. This is a good thing as it will notify us that something is not as we did expect.

```
In [6]: a, = (b for b in range(1))                                                            

In [7]: a                                                                                     
Out[7]: 0

In [8]: a, = (b for b in range(2))                                                            
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-c3f22dcf86ee> in <module>
----> 1 a, = (b for b in range(2))

ValueError: too many values to unpack (expected 1)
```

Reply:
Sorry, I don't understand the code.

Reply:
The code is using Pythons value unpacking to make sure the value we get is only one in the iterable. 

You can do this with tuples:
```
a, b, c = 1, 2, 3
```
which assigns `a` with 1 and `b` with 2, `c` with 3

Of course ths works with single-item tuples:
```
a, = [1]
```
this assings `a` with 1. But it has the advantage of failing when more items are being unpacked:

```
a, = 1,2
```
will fail with `ValueError: too many values to unpack (expected 1)`

This works for any iterable, including generators. So when you are tempting to do
```my_variable = some_iterable[0]```
and you assume only single item is available in `some_variable`, it is good to do
```my_variable, = some_iterable```
instead, because it will fail when the assumption is not met.

Reply:
Okay, got it. Thanks. Changed as you proposed.